selfie.c

```diff
@@ -3931,6 +3933,23 @@ void get_symbol() {
   }
 }
 
+void get_expected_symbol(uint64_t expected_symbol) {
+  if (symbol == expected_symbol)
+    get_symbol();
+  else
+    syntax_error_symbol(expected_symbol);
+}
+
+void get_required_symbol(uint64_t expected_symbol) {
```

Comment:
call `get_expected_symbol()` here but with a return value that indicates an error on which you then exit.